tools/convert_feat_to_wholegraph.py

```diff
@@ -16,160 +16,255 @@
+    # individual partitions. Then each feature is converted into the WholeGraph format.
+    if not use_low_mem:
+        # Read features from file
+        for path in (os.path.join(folder, name) for name in part_files):
+            feats_data.append(dgl.data.utils.load_tensors(f"{path}/{file_name}"))
+        num_parts = len(feats_data)
+        for type_name, feats in fname_dict.items():
+            for feat in feats:
+                feat = type_name + "/" + feat
+                if feat not in feats_data[0]:
+                    raise RuntimeError(
+                        f"Error: Unknown feature '{feat}'. Files contain \
+                                       the following features: {feats_data[0].keys()}."
+                    )
+                logging.info(f"Processing '{feat}' features...")
+                whole_feat_tensor = torch.concat(
+                    tuple(t[feat] for t in feats_data), dim=0
+                )
+                # Delete processed feature from memory
+                for t in feats_data:
+                    del t[feat]
+                wholegraph_processing(
+                    whole_feat_tensor, metadata, local_comm, feat, wg_folder, num_parts
+                )
+
+    # This version is less memory-consuming. For each feature, it iterates through all the
+    # partitions, loading the available features, but only stores (concatenating) only the
+    # current feature to be converted to WholeGraph.
+    else:  # low-mem
+        for ntype, feats in fname_dict.items():
+            for feat in feats:
+                feat = ntype + "/" + feat
+                node_feats_data = {feat: None}
+                num_parts = 0
+                # Read features from file
+                for path in (os.path.join(folder, name) for name in part_files):
+                    nfeat = dgl.data.utils.load_tensors(f"{path}/{file_name}")
+                    if feat not in nfeat:
+                        raise RuntimeError(
+                            f"Error: Unknown feature '{feat}'. Files contain \
+                                       the following features: {nfeat.keys()}."
+                        )
+                    if node_feats_data[feat] is None:
+                        node_feats_data[feat] = nfeat[feat]
+                    else:
+                        node_feats_data[feat] = torch.concat(
+                            (node_feats_data[feat], nfeat[feat]), dim=0
+                        )
+                    num_parts += 1
+                del nfeat
+                gc.collect()
+                wholegraph_processing(
+                    node_feats_data[feat],
+                    metadata,
+                    local_comm,
+                    feat,
+                    wg_folder,
+                    num_parts,
+                )
+
+        for path in (os.path.join(folder, name) for name in part_files):
+            feats_data.append(dgl.data.utils.load_tensors(f"{path}/{file_name}"))
```

Comment:
Are you loading all the feature data into memory again?

Reply:
This is not memory efficient. You should load one file using `dgl.data.utils.load_tensors(f"{path}/{file_name}")`, do L193-L197 and then load another one

Reply:
You are right! Since it does not consume 2X the amount of feature files, I skipped it. Fixed now.